spec/hydra/works/models/concerns/generic_file/characterization_spec.rb

```diff
@@ -0,0 +1,182 @@
+require 'spec_helper'
+
+# Mocking out the Hydra::Works::GenericFile sufficiently to add original_file
+# without a save to fedora. This works by mocking the response from ldp_source.head
+# so AF association believes subject is already saved.
+def mock_add_file_to_generic_file(gf, file)
+  allow(gf.ldp_source).to receive(:head).and_return(Faraday::Response.new)
+  allow(file).to receive(:has_content?).and_return(true)
+  gf.original_file = file
+end
+
+# Other fits to rdf mappings
+IMG_FITS_TO_RDF = { image_width:  ["imageWidth",  "toolname=Jhove"],
+                    image_height: ["imageHeight", "toolname=Jhove"],
+                    color_space:  ["colorSpace",  "toolname=Jhove"] }
+RT_FITS_TO_RDF = { jhove_time:     ["tool", "toolname=Jhove", "executionTime"],
+                   file_util_time: ["tool", "toolname=file utility", "executionTime"],
+                   exiftool_time:  ["tool", "toolname=Exiftool", "executionTime"],
+                   nlnz_time:      ["tool", "toolname=NLNZ Metadata Extractor", "executionTime"],
+                   ois_time:       ["tool", "toolname=OIS File Information", "executionTime"],
+                   ffident_time:   ["tool", "toolname=ffident", "executionTime"],
+                   tika_time:      ["tool", "toolname=Tika", "executionTime"] }
+
+describe Hydra::Works::GenericFile::Characterization do
+  let(:demo_class) do
+    Class.new(Hydra::Works::GenericFile::Base) do
+      include Hydra::Works::GenericFile::Characterization
+    end
+  end
+
+  # The is the only test context that will not mock out Characterization.characterize
+  #   Remove after development as this is more of an integration test
+  context "using fits.sh via Hydra::FileCharacterization" do
+    let(:filename)     { 'sample-file.pdf' }
+    let(:mimetype)     { 'application/pdf' }
+    let(:file_content) { File.open(File.join(fixture_path, filename)) }
+    let(:file)         { Hydra::PCDM::File.new.tap { |f| f.content = file_content } }
```

Comment:
You shouldn't need the `tap` here because `Hydra::PCDM::File` is a subclass of `ActiveFedora::File` which yields `self` to a block passed to the `initialize` method.